server/graphql/common/expenses.ts

```diff
@@ -786,6 +829,19 @@ export async function editExpense(
   );
 
   let payoutMethod = await expense.getPayoutMethod();
+
+  // Validate bank account payout method
+  if (payoutMethod?.type === PayoutMethodTypes.BANK_ACCOUNT) {
+    const payoutMethodData = <BankAccountPayoutMethodData>payoutMethod.data;
+    if (
+      !isAccountHolderNameAndLegalNameMatch(
+        payoutMethodData?.accountHolderName,
+        <string>expenseData.fromCollective.legalName,
+      )
+    ) {
```

Comment:
:ice_cream: I would personally keep the helper function just about comparing name (also to make it consistent with the frontend), and add a check here:


```suggestion
    const accountHolderName = payoutMethodData?.accountHolderName;
    const legalName = expenseData.fromCollective.legalName;
    if (accountHolderName && legalName && !namesAreEqual(accountHolderName, legalName)) {
```